react/src/components/ServiceLauncherModal.tsx

```diff
@@ -636,11 +763,64 @@ const ServiceLauncherModal: React.FC<ServiceLauncherProps> = ({
                       required
                     >
                       <Suspense fallback={<Skeleton.Input active />}>
-                        <VFolderLazyView uuid={endpoint?.model} />
+                        <VFolderLazyView
+                          uuid={endpoint?.model}
+                          mountDestination={endpoint?.model_mount_destination}
+                        />
                       </Suspense>
                     </Form.Item>
                   )
                 )}
+                <Flex
+                  direction="row"
+                  gap={'xxs'}
+                  align="stretch"
+                  justify="between"
+                >
+                  <Form.Item
+                    name={'modelMountDestination'}
+                    label={t('modelService.ModelMountDestination')}
+                    style={{ width: '50%' }}
+                    labelCol={{ style: { width: 400 } }}
+                  >
+                    <Input
+                      allowClear
+                      placeholder={'/models'}
+                      disabled={endpoint ? true : false}
+                    />
+                  </Form.Item>
+                  <MinusOutlined
+                    style={{ fontSize: token.fontSizeXL }}
+                    rotate={290}
+                  />
+                  <Form.Item
+                    name={'modelDefinitionPath'}
+                    label={t('modelService.ModelDefinitionPath')}
+                    style={{ width: '50%' }}
+                    labelCol={{ style: { width: 300 } }}
+                  >
+                    <Input
+                      allowClear
+                      placeholder={
+                        endpoint?.model_definition_path
+                          ? endpoint?.model_definition_path
+                          : 'model-definition.yaml'
+                      }
+                    />
+                  </Form.Item>
+                </Flex>
+                <VFolderTableFormItem
+                  rowKey={'id'}
+                  label={t('modelService.AdditionalMounts')}
+                  filter={(vf) =>
+                    vf.name !== selectedModelFolder &&
+                    vf.status === 'ready' &&
+                    vf.usage_mode !== 'model'
+                  }
+                  tableProps={{
+                    size: 'small',
+                  }}
+                />
               </>
```

Comment:
You don't need an extra state to update some component based on the form's value. Instead of using an extra state, you can wrap the part you want to update using `<Form.Item noStyle shouldUpdate ...>`. After that, you can delete the `selectedModelFolder` state, `updateSelectedModelFolder`, and the `useEffect` for that.

Some like this(I have not tested below code.):
```suggestion
                <Form.Item noStyle shouldUpdate={(prev,cur)=>prev.selectedModelFolder !== cur.selectedModelFolder}>
                  {()=>{
                    return <VFolderTableFormItem
                      rowKey={'id'}
                      label={t('modelService.AdditionalMounts')}
                      filter={(vf) =>
                        vf.name !== formRef.current?.getFieldValue('vFolderName') &&
                        vf.status === 'ready' &&
                        vf.usage_mode !== 'model'
                      }
                      tableProps={{
                        size: 'small',
                      }}
                    />
                  }}
                </Form.Item>
```